sentinel-core/src/main/java/com/alibaba/csp/sentinel/config/SentinelConfig.java

```diff
@@ -242,32 +237,54 @@ public static int statisticMaxRt() {
      * </li>
      * </ol>
      */
-    public static void resolveAppName() {
-
-        if (props.get(APP_NAME) != null) {
-            appName = props.get(APP_NAME);
+    private static void resolveAppName() {
+        // Priority: system env -> csp.sentinel.app.name -> project.name -> main class (or jar) name
+        String envKey = toEnvKey(APP_NAME_PROP_KEY);
+        String n = System.getenv(envKey);
+        if (!StringUtil.isBlank(n)) {
+            appName = n;
+            RecordLog.info("App name resolved from system env {}: {}", envKey, appName);
```

Comment:
The log info doesn't has prefix `[SentinelConfig]` like Line#71, and Line#71 move to inside `resolveAppType` method looks better. It's okay I think and we can do optimize them in the future.

Reply:
> The log info doesn't has prefix `[SentinelConfig]` like Line#71

Actually prefix is not strictly required if the log content is concrete enough :)

> Line#71 move to inside `resolveAppType` method looks better. It's okay I think and we can do optimize them in the future.

Good idea. We may improve this later.